wormhole-connect/src/hooks/useBalanceChecker.ts

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState, useMemo } from 'react';
+import { amount, routes } from '@wormhole-foundation/sdk';
+import { getWalletConnection, TransferWallet } from 'utils/wallet';
+import config from 'config';
+import { useIsMounted } from './useIsMounted';
+
+export const useBalanceChecker = (
+  quote?: routes.Quote<
+    routes.Options,
+    routes.ValidatedTransferParams<routes.Options>
+  >,
+): {
+  feeSymbol?: string;
+  isCheckingBalance: boolean;
+  isBalanceEnough: boolean;
+  walletBalance?: number;
+  networkCost?: number;
+} => {
+  const [isCheckingBalance, setCheckingBalance] = useState(false);
+  const [state, setState] = useState<{ balance: number; cost: number } | null>(
+    null,
+  );
+  const isMounted = useIsMounted();
+
+  const feeSymbol = useMemo(() => {
+    if (!quote?.relayFee?.token) return;
+
+    return config.sdkConverter.findTokenConfigV1(
+      quote?.relayFee?.token,
+      Object.values(config.tokens),
+    )?.symbol;
+  }, [quote]);
+
+  useEffect(() => {
+    setCheckingBalance(true);
+    (async () => {
+      try {
+        const wallet = getWalletConnection(TransferWallet.SENDING);
+        if (!wallet || !quote?.relayFee?.amount) return;
+
+        const cost = amount.whole(quote.relayFee.amount);
+        const balance = parseFloat(await wallet.getBalance());
+
+        if (isMounted.current) setState({ balance, cost });
+      } catch (e) {
+        console.error(e);
+        if (isMounted.current) setState(null);
+      } finally {
+        if (isMounted.current) setCheckingBalance(false);
```

Comment:
Is there a way to test this case? Like getting the isMounted hook unmounted while isCheckingBalance is true? It's a very very edge case I know, but do you think would it be possible to unmount the hook without the component still being mounted? In that case it might get stuck in checking state.

Reply:
it wouldn't be possible to unmount the hook while the compo is mounted in a real case. Hooks can't be conditional and they're bound together in the same render branch therefore hooks would only unmount along with the compo that called them.

That being said, a realistic way to test this (nvm fast fingers) would be to set an arbitrary long wait in try clause whilst unmounting through the UI (pressing the back button i think)